frontend/src/metabase/parameters/components/FilterApplyButton/FilterApplyButton.styled.ts

```diff
@@ -0,0 +1,40 @@
+import styled from "@emotion/styled";
+
+import { space } from "metabase/styled-components/theme";
+import Button from "metabase/core/components/Button/Button";
+
+export const TRANSITION_CLASSNAMES_PREFIX = "filter-apply-button";
+export const BUTTON_TRANSITION_DURATION = 200;
+
+export const ApplyButton = styled(Button)<{ isVisible: boolean }>`
+  margin-left: auto;
+  margin-top: ${space(1)};
+
+  opacity: ${({ isVisible }) => (isVisible ? 1 : 0)};
+  visibility: ${({ isVisible }) => (isVisible ? "visible" : "hidden")};
+
+  &,
+  &:hover {
+    transition-property: opacity, visibility;
+  }
+
+  &.${TRANSITION_CLASSNAMES_PREFIX}-enter {
```

Comment:
A bunch of these declarations are for transitioning the button when mounting and unmounting the component using `react-transition-group`.

You can test that when toggling the Auto-apply filters toggle on and off there should be a transition too.